nginx-controller/nginx/configurator.go

```diff
@@ -169,17 +217,69 @@ func (cnf *Configurator) generateNginxCfg(ingEx *IngressEx, pems map[string]stri
 			}
 		}
 
+		server.Maps = getVariableMaps(ingEx, rule)
+
 		if rootLocation == false && ingEx.Ingress.Spec.Backend != nil {
 			upsName := getNameForUpstream(ingEx.Ingress, emptyHost, ingEx.Ingress.Spec.Backend.ServiceName)
 			loc := createLocation(pathOrDefault("/"), upstreams[upsName], &ingCfg, wsServices[ingEx.Ingress.Spec.Backend.ServiceName], rewrites[ingEx.Ingress.Spec.Backend.ServiceName], sslServices[ingEx.Ingress.Spec.Backend.ServiceName])
 			locations = append(locations, loc)
 		}
 
 		server.Locations = locations
+
 		servers = append(servers, server)
 	}
 
-	return IngressNginxConfig{Upstreams: upstreamMapToSlice(upstreams), Servers: servers}
+	return IngressNginxConfig{
+		Upstreams: upstreamMapToSlice(upstreams),
+		Servers:   servers,
+	}
+}
+
+func getVariableMaps(ingEx *IngressEx, rule extensions.IngressRule) []Map {
+	maps := map[string]Map{}
+
+	for _, path := range rule.HTTP.Paths {
```

Comment:
there is no need to iterate rule.HTTP.Paths . Iterating over `ingEx.Endpoints` should be enough.
Also, this function is complicated. It is better to split into multiple functions.

Reply:
First part is fixed, regarding second part — I doubt that there's good way to split it into two functions without loosing cohesion. Method is only 40 lines long.